sdk/storage/azure-storage-file/src/test/java/com/azure/storage/file/spock/ShareAPITests.groovy

```diff
@@ -0,0 +1,350 @@
+        primaryShareClient.create()
+        when:
+        def fileClient = primaryShareClient.getFileClient("testFile")
+        fileClient.getProperties()
+        then:
+        assertNotNull(fileClient)
+        def e = thrown(HttpResponseException)
+        assertEquals(e.response().statusCode(), 404)
+    }
+
+    def "Create share from share client"() {
+        expect:
+        FileTestHelper.assertResponseStatusCode(primaryShareClient.create(), 201)
+    }
+
+    @Unroll
+    def "Create share with args from share client"() {
+        expect:
+        FileTestHelper.assertResponseStatusCode(primaryShareClient.create(metadata, quota), 201)
+        where:
+        metadata     | quota
+        null         | null
+        null         | 1
+        testMetadata | null
+        testMetadata | 1
+    }
+
+    @Unroll
+    def "Create share with invalid args from share client"() {
+        when:
+        primaryShareClient.create(metadata, quota)
+        then:
+        def e = thrown(StorageErrorException)
+        FileTestHelper.assertExceptionStatusCodeAndMessage(e, statusCode, errMessage)
+        where:
+        metadata                                 | quota | statusCode | errMessage
+        Collections.singletonMap("", "value")    | 1     | 400        | StorageErrorCode.EMPTY_METADATA_KEY
+        Collections.singletonMap("a@B", "value") | 1     | 400        | "Bad Request"
+        testMetadata                             | 6000  | 400        | StorageErrorCode.INVALID_HEADER_VALUE
+    }
+
+    def "Create snapshot from share client"() {
+        given:
+        primaryShareClient.create()
+        def shareSnapshotName = testResourceName.randomName("share", 16)
+        when:
+        def createSnapshotResponse = primaryShareClient.createSnapshot()
+        def shareSnapshotClient = new ShareClientBuilder().shareName(shareSnapshotName).connectionString(connectionString)
+            .snapshot(createSnapshotResponse.value().snapshot()).buildClient()
+        then:
+        assertEquals(createSnapshotResponse.value().snapshot(),
+            shareSnapshotClient.getSnapshotId())
+    }
+
+    def "Create snapshot error from share client"() {
+        when:
+        primaryShareClient.createSnapshot()
+        then:
+        def e = thrown(StorageErrorException)
+        FileTestHelper.assertExceptionStatusCodeAndMessage(e, 404, StorageErrorCode.SHARE_NOT_FOUND)
+    }
+
+    def "Create snapshot metadata from share client"() {
+        given:
+        primaryShareClient.create()
+        def shareSnapshotName = testResourceName.randomName("share", 16)
+        when:
+        def createSnapshotResponse = primaryShareClient.createSnapshot(testMetadata)
+        def shareSnapshotClient = new ShareClientBuilder().shareName(shareSnapshotName).connectionString(connectionString)
+            .snapshot(createSnapshotResponse.value().snapshot()).buildClient()
+        then:
+        assertEquals(createSnapshotResponse.value().snapshot(),
```

Comment:
Instead lets check that the metadata on the snapshot is what was set, this check doesn't really test anything useful.